sdk/mobile-center-updates/src/main/java/com/microsoft/azure/mobile/updates/Updates.java

```diff
@@ -0,0 +1,971 @@
+        return LOG_TAG;
+    }
+
+    @Override
+    public synchronized void onStarted(@NonNull Context context, @NonNull String appSecret, @NonNull Channel channel) {
+        super.onStarted(context, appSecret, channel);
+        mContext = context;
+        mAppSecret = appSecret;
+        resumeUpdateWorkflow();
+    }
+
+    @Override
+    public synchronized void onActivityCreated(Activity activity, Bundle savedInstanceState) {
+
+        /* Resolve launcher class name only once, use empty string to cache a failed resolution. */
+        if (mLauncherActivityClassName == null) {
+            mLauncherActivityClassName = "";
+            PackageManager packageManager = activity.getPackageManager();
+            Intent intent = packageManager.getLaunchIntentForPackage(activity.getPackageName());
+            if (intent != null) {
+                mLauncherActivityClassName = intent.resolveActivity(packageManager).getClassName();
+            }
+        }
+
+        /* Clear workflow finished state if launch recreated, to achieve check on "startup". */
+        if (activity.getClass().getName().equals(mLauncherActivityClassName)) {
+            MobileCenterLog.info(LOG_TAG, "Launcher activity restarted.");
+            if (StorageHelper.PreferencesStorage.getString(PREFERENCE_KEY_DOWNLOAD_URI) == null) {
+                mWorkflowCompleted = false;
+                mBrowserOpened = false;
+            }
+        }
+    }
+
+    @Override
+    public synchronized void onActivityResumed(Activity activity) {
+        mForegroundActivity = activity;
+        resumeUpdateWorkflow();
+    }
+
+    @Override
+    public synchronized void onActivityPaused(Activity activity) {
+        mForegroundActivity = null;
+    }
+
+    @Override
+    public synchronized void setInstanceEnabled(boolean enabled) {
+        super.setInstanceEnabled(enabled);
+        if (enabled) {
+            resumeUpdateWorkflow();
+        } else {
+
+            /* Clean all state on disabling, cancel everything. */
+            mBrowserOpened = false;
+            mWorkflowCompleted = false;
+            cancelPreviousTasks();
+            StorageHelper.PreferencesStorage.remove(PREFERENCE_KEY_UPDATE_TOKEN);
+            StorageHelper.PreferencesStorage.remove(PREFERENCE_KEY_REQUEST_ID);
+        }
+    }
+
+    /**
+     * Implements {@link #setLoginUrl(String)}.
+     */
+    private synchronized void setInstanceLoginUrl(String loginUrl) {
+        mLoginUrl = loginUrl;
+    }
+
+    /**
+     * Implements {@link #setApiUrl(String)}}.
+     */
+    private synchronized void setInstanceApiUrl(String apiUrl) {
+        mApiUrl = apiUrl;
+    }
+
+    /**
+     * Cancel everything.
+     */
+    private synchronized void cancelPreviousTasks() {
+        if (mCheckReleaseApiCall != null) {
+            mCheckReleaseApiCall.cancel();
+            mCheckReleaseApiCall = null;
+            mCheckReleaseCallId = null;
+        }
+        mUpdateDialog = null;
```

Comment:
We already talked about that and you said it's bad to hide the dialog while is looking at it. Buttons won't do anything though.